src/strings.c

```diff
@@ -1977,6 +1977,8 @@ f_trim(typval_T *argvars, typval_T *rettv)
     if (argvars[1].v_type == VAR_STRING)
     {
 	mask = tv_get_string_buf_chk(&argvars[1], buf2);
+	if (strlen(mask) == 0)
```

Comment:
Need to check for NULL (e.g. when passing a Dictionary which causes a type error)

Reply:
The type of the argument is checked in the `if` statement two lines above.
It can only be `VAR_STRING`.

Reply:
Oops, didn't notice that.

Reply:
No worries.
Thank you for looking into it!